addons/website/static/src/snippets/s_table_of_content/options.js

```diff
@@ -50,9 +71,29 @@ options.registry.TableOfContent = options.Class.extend({
         const $nav = this.$target.find('.s_table_of_content_navbar');
         const $headings = this.$target.find(this.targetedElements);
         $nav.empty();
+        const uniqueHeadingIds = new Set();
         _.each($headings, el => {
             const $el = $(el);
-            const id = 'table_of_content_heading_' + _.now() + '_' + _.uniqueId();
+            if (!this.$target[0].dataset.tocId) {
+                const tocEls = this.$target[0].closest("#wrapwrap").querySelectorAll("section.s_table_of_content");
```

Comment:
Nitpicking:
- I don't like new references to the `#wrapwrap` element, maybe just search in the whole DOM? In 16.0 it makes sense and in 15.0, it's ok if you found other table of content which are not really in the page (like in the snippets menu).
- Instead of `section.s_table_of_content`, search using `[data-snippet]`, I'd like to progressively stop using anything else in javascript code and python code when searching for snippets.